app/views/2017/shared/_header.html.erb

```diff
@@ -15,8 +15,15 @@
       <li class="nav-link"><%= link_to t("header.search"),  "/search" %></li>
       <li class="nav-link"><%= link_to t("header.about"),   "/about" %></li>
     </ul>
+    <% unless I18n.locale == I18n.default_locale %>
+      <ul>
+        <% @locale = LocaleService.find(locale: nil, lang_code: I18n.locale) %>
```

Comment:
Can we move this to a `before_action` method in `ApplicationController`?

Maybe something like `set_current_locale`?

Reply:
I'm not a big fan of callback like `before_action` and other since they have tendency to make testing harder and I think we should prefer extracting logic in service and / or query object instead of relying  on callback, but if you really want it to be in a callback, I'll refactor it. :)